src/FlashList.tsx

```diff
@@ -126,10 +126,14 @@ class FlashList<T> extends React.PureComponent<
       throw new CustomError(ExceptionList.columnsWhileHorizontalNotSupported);
     }
 
+    const styles = Array.isArray(this.props.style) ? this.props.style : [this.props.style]
+
     // `createAnimatedComponent` always passes a blank style object. To avoid warning while using AnimatedFlashList we've modified the check
-    if (Object.keys(this.props.style || {}).length > 0) {
+    // `style` prop can be an array. So we need to validate every object in array. Check: https://github.com/Shopify/flash-list/issues/651
+    if(styles.some((e) => Object.keys(e || {}).length > 0)) {
```

Comment:
We can use [flatten](https://reactnative.dev/docs/stylesheet#flatten) instead. [This](https://github.com/Shopify/flash-list/pull/654/files#diff-f8655d3e55acbb01f8cca2007bd8e740be197492eecb49d145dbdf476a423b0fR129) line can then be reverted. This will also work for `style={[[{}]]}` or whatever combination of empty arrays you would use.

Since this check is relevant only in `DEV`, we should also add a check for that first.


```suggestion
    if (__DEV__ && Object.keys(StyleSheet.flatten(this.props.style ?? {})).length > 0) {
```